src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,202 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTDATA = {};
+    private int size;
+    private Object[] elementData;
+
+    public ArrayList() {
+        this.elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+    }
 
+    public boolean add(T e) {
+        add(e, elementData, size);
+        return true;
+    }
+
+    private void add(T e, Object[] elementData, int s) {
+        if (s == elementData.length) {
+            elementData = grow();
+        }
+        elementData[s] = e;
+        size = s + 1;
     }
 
     @Override
     public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        final int s;
+        Object[] elementData;
+        if ((s = size) == (elementData = this.elementData).length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                s - index);
+        elementData[index] = value;
+        size = s + 1;
+    }
 
+    public Object[] toArray(List<T> list) {
+        Object[] array = new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            array[i] = list.get(i);
+        }
+        return array;
     }
 
     @Override
     public void addAll(List<T> list) {
+        Object[] a = toArray(list);
+        int numNew = a.length;
+        if (numNew == 0) {
+            return;
+        }
+        Object[] elementData;
+        final int s;
+        if (numNew > (elementData = this.elementData).length - (s = size)) {
+            elementData = grow(s + numNew);
+        }
+        System.arraycopy(a, 0, elementData, s, numNew);
+        size = s + numNew;
+    }
 
+    private void checkIndex(int index) {
+        if ((index >= size || index < 0)) {
+            throw new ArrayListIndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return elementData(index);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        final Object[] es = elementData;
+        @SuppressWarnings("unchecked") T oldValue = (T) es[index];
+        fastRemove(es, index);
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
```

Comment:
rewrite this method using 1 loop and 1 IF inside.
Never use **break label** in your code, it is bad practice.